fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/physical/PhysicalJoinOperator.java

```diff
@@ -34,6 +33,7 @@ public abstract class PhysicalJoinOperator extends PhysicalOperator {
     protected final JoinOperator joinType;
     protected final ScalarOperator onPredicate;
     protected final String joinHint;
+    protected boolean canShuffleOutput = false;
```

Comment:
move to fragment builder

Reply:
AddDecodeNodeRule will create new PhysicalJoinOperator which may cause this optimize failed

Reply:
OK. thanks @Seaven and @stdpain . fixed. code moved to fragment builder.